Src/BootCamp.Chapter/TextTable.cs

```diff
@@ -30,11 +33,141 @@ public static class TextTable
         /// <summary>
         /// Build a table for given message with given padding.
         /// Padding means how many spaces will a message be wrapped with.
-        /// Table itself is made of: "+-" symbols. 
+        /// Table itself is made of: "+-" symbolmessage. 
         /// </summary>
         public static string Build(string message, int padding)
         {
-            return "";
+            //Analyze message
+            if (String.IsNullOrEmpty(message))
+            {
+                return "";
+            }
+            var numOfMessageLines = GetNumberOfMessageLines(message);
+            var messageLine = GetMessageLines(message, numOfMessageLines);
+            var maxLength = GetMaxMessageLineLength(messageLine);
+
+            //Build table from message
+            return BuildTable(messageLine, maxLength, padding);
+        }
+
+        private static int GetNumberOfMessageLines(string message)
+        {
+            var numOfLines = 1;
+
+            for (var i = 0; i < message.Length - 1; i++)
+            {
+                if (message[i] == '\r' && message[i + 1] == '\n')
+                {
+                    numOfLines++;
+                }
+            }
+
+            return numOfLines;
+        }
+
+        private static string[] GetMessageLines(string message, int numOfMessageLines)
+        {
+            var messageLine = new string[numOfMessageLines];
+            var messageCount = 0;
+            var beginPos = 0;
+            var endPos = 0;
+
+            for (var i = 0; i < message.Length - 1; i++)
+            {
+                if (message[i] == '\r' && message[i + 1] == '\n')
+                {
+                    endPos = i - 1;
+                    messageLine[messageCount++] = message.Substring(beginPos, endPos - beginPos + 1);
+                    beginPos = i + 2;
+                }
+            }
+
+            if (numOfMessageLines == 0 && message.Length > 0)
+            {
+                messageLine[0] = message;
+            }
+
+            if (endPos <= message.Length)
+            {
+                messageLine[messageCount++] = message.Substring(beginPos, message.Length - beginPos);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'messageCount'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4955013)